better error message when offer/answer get confused

I accidentally typed `conn.SetRemoteDescription(offer)` when I should have written `conn.SetRemoteDescription(answer)`. This resulted in a nil pointer panic. It took me a while digging around in the code to find out what went wrong. It would be nice if pion could detect this mistake and return a helpful error message instead of panic.

I can confirm this.
```
--- FAIL: TestSetRemoteDescription (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7ac79d]

goroutine 19 [running]:
testing.tRunner.func1.1(0x822540, 0xc00870)
        /usr/lib/go-1.14/src/testing/testing.go:940 +0x2f5
testing.tRunner.func1(0xc0000d0d80)
        /usr/lib/go-1.14/src/testing/testing.go:943 +0x3f9
panic(0x822540, 0xc00870)
        /usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
github.com/pion/webrtc/v2.(*PeerConnection).SetRemoteDescription(0xc0000ada00, 0x1, 0xc000174500, 0x248, 0xc0000d0fc0, 0x0, 0x0)
        /home/jch/go/pkg/mod/github.com/pion/webrtc/v2@v2.2.14/peerconnection.go:796 +0x87d
sfu.TestSetRemoteDescription(0xc0000d0d80)
        /home/jch/go/src/sfu/sra_test.go:27 +0x2fc
testing.tRunner(0xc0000d0d80, 0x8ca568)
        /usr/lib/go-1.14/src/testing/testing.go:991 +0xdc
created by testing.(*T).Run
        /usr/lib/go-1.14/src/testing/testing.go:1042 +0x357
exit status 2
```
Here's the test file:
```
package main

import (
	"testing"

	"github.com/pion/webrtc/v2"
)

func TestSetRemoteDescription(t *testing.T) {
	api := webrtc.NewAPI()
	pc, err := api.NewPeerConnection(webrtc.Configuration{})
	if err != nil {
		t.Fatalf("NewPeerConnection: %v", err)
	}
	defer pc.Close()

	offer, err := pc.CreateOffer(nil)
	if err != nil {
		t.Fatalf("CreateOffer: %v", err)
	}

	err = pc.SetLocalDescription(offer)
	if err != nil {
		t.Fatalf("SetLocalDescription: %v", err)
	}

	err = pc.SetRemoteDescription(offer)
	if err == nil {
		t.Fatalf("SetRemoteDescription: got nil")
	}
}
```

--- a/peerconnection.go
+++ b/peerconnection.go
@@ -881,7 +881,7 @@ func (pc *PeerConnection) setDescription(sd *SessionDescription, op stateChangeO
 			return nextState, &rtcerr.OperationError{Err: fmt.Errorf("%w: %q", errPeerConnStateChangeUnhandled, op)}
 		}
 
-		return nextState, nil
+		return nextState, err
 	}()
 
 	if err == nil {

--- a/peerconnection_test.go
+++ b/peerconnection_test.go
@@ -636,3 +636,19 @@ func TestGatherOnSetLocalDescription(t *testing.T) {
 	assert.NoError(t, pcOffer.Close())
 	assert.NoError(t, pcAnswer.Close())
 }
+
+// Assert that SetRemoteDescription handles invalid states
+func TestSetRemoteDescriptionInvalid(t *testing.T) {
+	t.Run("local-offer+SetRemoteDescription(Offer)", func(t *testing.T) {
+		pc, err := NewPeerConnection(Configuration{})
+		assert.NoError(t, err)
+
+		offer, err := pc.CreateOffer(nil)
+		assert.NoError(t, err)
+
+		assert.NoError(t, pc.SetLocalDescription(offer))
+		assert.Error(t, pc.SetRemoteDescription(offer))
+
+		assert.NoError(t, pc.Close())
+	})
+}

